Eb/ 이슈 생성 page 구현

Eunbin-Kim opened the pull request:
### 이슈 생성 page 구현
- 레이아웃에 맞게 이슈 생성 page 내 컴포넌트 조합
- labels, milestone API 요청하여, 버튼 클릭 시 기본 데이터 modal 창 뜨도록 구현
- labels: 클릭함에 따라서 태그들 추가됨
- milestone: 클릭 후 상태 변경되나 view에 반영되지 않음(이유를 찾지 못함...)

+) title 추가 필요
+) create issue 버튼에 따른 API 요청 보내기
+) milestone 하나만 선택되도록 수정
+) modal 창 재랜더링에 따른 선택 초기화 수정

Eunbin-Kim committed:
[feat] label store 구현
- API 요청하여 label 읽어오고 store 저장 구현
- API 요청할 수 있는 dispatch, action 구현

Eunbin-Kim committed:
[feat] milestone store 구현
- GET milestone API로 milestone store 구현
- API 요청할 수 있는 action, dispatch 구현

Eunbin-Kim committed:
[feat] label, milestone 전역 상태 지정
- label, milestone 전역 상태로 지정하여 provider 설정

Eunbin-Kim committed:
[refactor] issue store 디렉토리 생성
- label, milestone store 생성함에 따라 각각 디렉토리 생성하여 관리

Eunbin-Kim committed:
[feat] modal 내 milestone item component 구현
- milestone item component 구현

Eunbin-Kim committed:
[feat] 이슈생성 page 구현
- 이슈 생성 page view 구현
- labels, milestone modal 창 구현

Eunbin-Kim committed:
[feat] 이슈생성 page index.ts 추가

Eunbin-Kim committed:
[fix] useAsync 다양한 타입 수용가능하도록 수정
- 타입 문제로 기존 만들어놓은 API 에서만 작동하는 한계 발생
- generic 지정하여 다양한 타입 수용가능하도록 수정

Eunbin-Kim committed:
[feat] store 내 issue 디렉토리 추가로 인한 수정
- store 내 issue 디렉토리 추가함에 따라 경로 수정 및 기존 index 파일 수정

cyjo9603 commented on client/src/stores/label/index.ts:
```diff
@@ -0,0 +1,48 @@
+  }
+};
+
+export default createContext<LabelsContext>({
+  ...initialLabels,
+  dispatch: () => {},
+});
```
context store 만든게 완벽하네요!

cyjo9603 commented on client/src/stores/label/readLabels.ts:
```diff
@@ -0,0 +1,44 @@
+import { Label } from '@stores/type';
+
+export const READ_LABELS_REQUEST = 'READ_LABELS_REQUEST';
+export const READ_LABELS_SUCCESS = 'READ_LABELS_SUCESS';
```
`READ_LABELS_SUCESS`에 오타가 있네요^^ 수정 부탁드립니다

cyjo9603 commented on client/src/stores/milestone/readMilestones.ts:
```diff
@@ -0,0 +1,44 @@
+import { Milestone } from '@stores/type';
+
+export const READ_MILESTONES_REQUEST = 'READ_MILESTONES_REQUEST';
+export const READ_MILESTONES_SUCCESS = 'READ_MILESTONES_SUCESS';
```
`READ_MILESTONES_SUCESS` 오타를 또 찾아버렸네요^^ 수정 부탁드려요 😊

cyjo9603 commented on client/src/app.tsx:
```diff
@@ -1,26 +1,80 @@
+    milestoneReducer,
+    initialMilestones,
+  );
+  const asyncIssueDispatch = useMemo(
+    () => useAsync<IssueActions>(issueDispatch),
+    [issueDispatch],
+  );
+  const asyncLabelDispatch = useMemo(
+    () => useAsync<LabelActions>(labelDispatch),
+    [labelDispatch],
+  );
+  const asyncMilestoneDispatch = useMemo(
+    () => useAsync<MilestoneActions>(milestoneDispatch),
+    [milestoneDispatch],
+  );
```
최적화 작업 조흔것 같습니당

cyjo9603 commented on client/src/components/molecules/SelectMenuModal/SelectMenuItem/SelectMenuItemMilestone/SelectMenuItemMilestone.tsx:
```diff
@@ -0,0 +1,45 @@
+  isClicked: boolean;
+}
+
+const StyledLi = styled.li<StyledProps>`
+  background-color: ${({ isClicked, theme }) =>
+    isClicked && theme.palette.BG_COLOR02};
+`;
```
isCliked를 styled 안에서 관리하는 것보다 css class를 사용해 관리하는게 더 좋아보이네용

cyjo9603 reviewed (changes requested):
전체적으로 잘 작성하신것 같아요! 코멘트 드린 부분중에 몇 부분만 수정해 주시면 될것 같습니다

Eunbin-Kim committed:
[fix] 리스트 내 아이템 클릭 기능 수정
- 리스트 내 아이템 클릭 후 상태값 저장되나 화면에 표시되지 않는 버그 수정
- 클릭된 아이템 className으로 구분하도록 수정
- 모든 요소 동일한 타입으로 동작하도록 수정

Eunbin-Kim committed:
[fix] 리스트 선택된 아이템 상태관리 로직 수정
- onClick에 따라서 선택된 아이템 상태에 추가하여 재랜더링
- 기존 하위 컴포넌트에서 따로 상태값 사용하던 것 삭제

Eunbin-Kim committed:
[feat] label tag class명 추가
- margin 조정을 위한 class명 추가

Eunbin-Kim committed:
[fix] title 표시 오류 수정
- title 초기 값 빈 스트링으로 들어가기 때문에 undefined 일 경우에만 없애는 것으로 수정

Eunbin-Kim committed:
[refactor] write form style 수정
-  write form 반응형 고려하여 style 수정

Eunbin-Kim committed:
[refactor] 페이지 마진 일부 조정(style)

Eunbin-Kim committed:
[chore] SUCCESS 오타 수정

cyjo9603 reviewed (approved):
👍👍👍
